chain/chain/src/resharding.rs

```diff
@@ -317,25 +324,49 @@ impl Chain {
         let checked_account_id_to_shard_uid =
             get_checked_account_id_to_shard_uid_fn(shard_uid, new_shards, next_epoch_shard_layout);
 
+        let shard_uid_string = shard_uid.to_string();
+        let metrics_labels = [shard_uid_string.as_str()];
+
         // Once we build the iterator, we break it into batches using the get_trie_update_batch function.
-        while let Some(batch) = get_trie_update_batch(&mut iter) {
+        loop {
+            // Prepare the batch.
+            let batch = {
+                let histogram = RESHARDING_BATCH_PREPARE_TIME.with_label_values(&metrics_labels);
+                let _timer = histogram.start_timer();
+                let Some(batch) = get_trie_update_batch(&config, &mut iter) else { break };
+                batch
+            };
+
+            // Apply the batch - add values to the split states.
             let TrieUpdateBatch { entries, size } = batch;
-            // TODO(#9435): This is highly inefficient as for each key in the batch, we are parsing the account_id
-            // A better way would be to use the boundary account to construct the from and to key range for flat storage iterator
-            let (store_update, new_state_roots) = tries.add_values_to_split_states(
-                &state_roots,
-                entries,
-                &checked_account_id_to_shard_uid,
-            )?;
-            state_roots = new_state_roots;
-            store_update.commit()?;
-            RESHARDING_BATCH_COUNT.with_label_values(&[shard_uid.to_string().as_str()]).inc();
-            RESHARDING_BATCH_SIZE
-                .with_label_values(&[shard_uid.to_string().as_str()])
-                .add(size as i64)
+            let store_update = {
+                let histogram = RESHARDING_BATCH_APPLY_TIME.with_label_values(&metrics_labels);
+                let _timer = histogram.start_timer();
+                // TODO(#9435): This is highly inefficient as for each key in the batch, we are parsing the account_id
+                // A better way would be to use the boundary account to construct the from and to key range for flat storage iterator
+                let (store_update, new_state_roots) = tries.add_values_to_split_states(
+                    &state_roots,
+                    entries,
+                    &checked_account_id_to_shard_uid,
+                )?;
+                state_roots = new_state_roots;
+                store_update
+            };
+
+            // Commit the store update.
+            {
+                let histogram = RESHARDING_BATCH_COMMIT_TIME.with_label_values(&metrics_labels);
+                let _timer = histogram.start_timer();
+                store_update.commit()?;
+            }
+
+            RESHARDING_BATCH_COUNT.with_label_values(&metrics_labels).inc();
+            RESHARDING_BATCH_SIZE.with_label_values(&metrics_labels).add(size as i64);
+            std::thread::sleep(config.batch_delay);
```

Comment:
optional: quick comment as to why we have this sleep (to ease out writes to RocksDB which is in contention with block processing..)